### DISCUSSION TITLE
Milvus Metadata Management

### DISCUSSION THREAD

**Discussion by Pranav-NG** [milvus-io/milvus#27242] [Q&A and General discussion]
I came across this pose for Metadata Management.
https://milvus.io/blog/managing-metadata-in-milvus-1.md

Can you please explain how to configure this with latest version of Milvus and Python SDK.

When metadata is stored in MySQL , does it have an id to link back to text chunks in vector DB . ?

**Comment by yhmo** [comment]
This article is for the earlier version v1.x, 4 years ago.
The latest version is v2.3, all the v2.x versions store metadata in an Etcd service.
If you deploy a milvus standalone locally, you will see three containers:
```
CONTAINER ID   NAME                CPU %     MEM USAGE / LIMIT     MEM %     NET I/O         BLOCK I/O         PIDS
34eb1077088b   milvus-etcd         0.44%     25.86MiB / 62.66GiB   0.04%     157MB / 104MB   12.7MB / 1.04GB   19
c4e3313049e4   milvus-minio        0.01%     108.7MiB / 62.66GiB   0.17%     1.2MB / 168kB   68.1MB / 38.9MB   18
3a19dbf147d7   milvus-standalone   2.85%     200MiB / 62.66GiB     0.31%     105MB / 156MB   11.8MB / 13.9MB   47
```
Etcd stores metadata, MinIO stores data files.

If you enter the milvus-standalone container by `docker exec -it [containerID] bash`, you will see a "milvus" folder under the root path.
There is a milvus.yaml file under the path /milvus/configs/
```
root@3a19dbf147d7:/milvus# tree -L 2
.
|-- bin
|   `-- milvus
|-- configs
|   |-- OWNERS
|   |-- advanced
|   |-- cert
|   |-- glog.conf
|   |-- hook.yaml
|   `-- milvus.yaml
`-- lib
```
The milvus.yaml is the configuration file of Milvus, you can read this doc to know all the configurations: https://milvus.io/docs/configure-docker.md. And this doc also tells how to configure milvus v2.x.

**Reply by Pranav-NG** [reply]
Thank you for the quick response. The above answer makes sense. '

So , Is it possible to have connection with MySQL for metadata in this version ?

**Reply by yhmo** [reply]
> Thank you for the quick response. The above answer makes sense. '
> 
> So , Is it possible to have connection with MySQL for metadata in this version ?

No, Milvus v2.x doesn't support using MySQL as metadata store. We did try MySQL to store metadata but we found its performance cannot meet the requirement of large amount of interactions for metadata.
